src/Simplex/Messaging/Agent.hs

```diff
@@ -312,9 +351,26 @@ acceptInv c connId invId connInfo =
 subscribeConnection' :: AgentMonad m => AgentClient -> ConnId -> m ()
 subscribeConnection' c connId =
   withStore (`getConn` connId) >>= \case
-    SomeConn _ (DuplexConnection _ rq _) -> subscribeQueue c rq connId
+    SomeConn _ (DuplexConnection _ rq sq) -> case status (sq :: SndQueue) of
+      Confirmed -> do
+        let verifyKey = C.publicKey $ sndPrivateKey sq
+        case verifyKey of
+          Nothing -> throwError $ CONN NO_PUBLIC_KEY
+          Just vk -> do
+            activateQueueInitiating c connId sq vk 5 0
+            subscribeQueue c rq connId
+      Active -> subscribeQueue c rq connId
+      _ -> throwError $ CONN BAD_QUEUE_STATUS
```

Comment:
ditto here

Reply:
also, we don't generally validate that the queue is in the expected status, so it could be just `_ -> subscribeQueue c rq connId`

Reply:
done